### PR TITLE
[auth0-js] Add optional `connection` property to `Popup.authorize` options

### PR DESCRIPTION
**Pull request by timswalling**
PR #20852 changed the type of `Popup.authorize` options from `any` to an object literal, but omitted the optional `connection` property.

This PR adds `connection` as an optional string.

---
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://auth0.com/docs/libraries/auth0js/v8#webauth-popup-authorize-

### CODE REVIEW

**Review by adrianchia — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
